### PR TITLE
arm64: cache: refine arch_dcache_range()

### PR DESCRIPTION
**Pull request by JiafeiPan**
Move all assembly code to c code.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

### CODE REVIEW

File: arch/arm64/core/cache.c [hunk 1]
```diff
@@ -39,20 +39,43 @@
...
+	uintptr_t start_addr = (uintptr_t)addr;
+	uintptr_t end_addr = start_addr + size;
+
+	if (op != K_CACHE_INVD && op != K_CACHE_WB && op != K_CACHE_WB_INVD)
 		return -ENOTSUP;
-	}
+
+	ctr_el0 = read_sysreg(CTR_EL0);
+	dminline = (ctr_el0 >> CTR_EL0_DMINLINE_SHIFT) & CTR_EL0_DMINLINE_MASK;
+	line_size = 4 << dminline;
```
Comment (carlocaione) [inline arch/arm64/core/cache.c:63]: Why not directly using `arch_dcache_line_size_get()`?

Comment (JiafeiPan) [reply arch/arm64/core/cache.c:63]: I am not sure why use the following code to get line size in arch_dcache_line_size_get()?

dcache_line_size = cwg ? 4 << cwg : 4 << dminline;

I find it use 4 << DminLine to get the mininual line size in  sample code of "Cleaning by Virtual Address" in cortex a-v8 programer guider.

Comment (JiafeiPan) [reply arch/arm64/core/cache.c:63]: @MrVan , could you please address why use cwg field to get dcache line size, I found in the other SW component such as tf-a and uboot, and also in ARM's sample code, it only get the minimal dcache line size by using dminline ? thanks.

**Comment by npitre** [comment]
There was a `dsb` at the end of the loop in the assembly version.
Don't you need it in the C version as well?

File: arch/arm64/core/cache.c (hunk 1 above)
Comment (carlocaione) [reply arch/arm64/core/cache.c:63]: Indeed not clear what the CWG is for there.

Comment (JiafeiPan) [reply arch/arm64/core/cache.c]: Yes, I removed CSW and use dminline only.

**Comment by JiafeiPan** [comment]
> There was a `dsb` at the end of the loop in the assembly version.
> Don't you need it in the C version as well?

Yes,  we need dsb there, updated.

File: arch/arm64/core/cache.c (hunk 1 above)
Comment (JiafeiPan) [reply arch/arm64/core/cache.c]: Typo, removed CWG.

**Review by carlocaione — APPROVED** [review]

**Review by npitre — APPROVED** [review]

**Pull request merged**